### ISSUE TITLE
希望可以增加用户权限相关的openapi

### ISSUE THREAD

**Issue by uucloud** [apolloconfig/apollo#5265]
**你的特性请求和某个问题有关吗？请描述**

最新的2.3.0支持了创建应用，可以自动化和公司内部应用对接了，但是只有创建应用时可以配置管理员和负责人，希望可以增加用户权限相关的api，这样权限变更也可以自动对齐

**Comment by ydbdyds** [comment]
+1

**Comment by habiibullahm** [comment]
Hi @nobodyiam — I'd like to discuss / help with this if it's still open.

This looks closely related to #5099 (user management OpenAPI). After create-app OpenAPI landed, permission changes still seem hard to automate without Portal UI flows.

Proposed approach (after aligning with #5099):
1. Define OpenAPI for app/user role assignment & query (grant/revoke/list)
2. Implement against existing Portal permission services with token-based auth
3. Keep the contract stable for SSO/ops automation use cases

Questions before coding:
- Should this be folded into #5099 as one OpenAPI permissions effort, or kept as a follow-up PR?
- Is this free to claim, and which permission APIs are in-scope for v1?

Thanks!

**Comment by nobodyiam** [comment]
Thanks for checking before coding. This should remain separate from #5099 conceptually: #5099 is about user discovery and user management, while this issue is about the post-creation app and namespace permission lifecycle. However, the requested API surface is already implemented on `master`, so there is no new implementation PR to claim.

[PR #5617](https://github.com/apolloconfig/apollo/pull/5617) added generated `PermissionManagementApi` endpoints backed by the existing Portal permission services. The contract includes role-user queries plus grant and revoke operations for app, namespace, environment-namespace, and cluster-namespace scopes. For example:

- `GET /openapi/v1/apps/{appId}/role-users`
- `POST /openapi/v1/apps/{appId}/roles/{roleType}`
- `DELETE /openapi/v1/apps/{appId}/roles/{roleType}`

The namespace-scoped variants are part of the same [Permission Management contract](https://github.com/apolloconfig/apollo-openapi/blob/v0.3.10/apollo-openapi.yaml#L3880-L4469). Mutating endpoints retain the existing app-scoped authorization checks and operator audit attribution. [PR #5632](https://github.com/apolloconfig/apollo/pull/5632) additionally exposes these operations to scoped user access tokens through `app:manage-role`.

One boundary is worth keeping explicit: an app owner is application metadata, not a generic role. Changing it uses `PUT /openapi/v1/apps/{appId}`, while administrator, modify, and release permissions use the Permission Management endpoints.

These changes are listed for Apollo 3.0.0 and are not included in Apollo 2.5.2. Please do not fold this into a new #5099 implementation PR. A useful follow-up contribution would be to validate current `master` against the original permission-sync workflow or fill any remaining documentation and test gaps.

**Comment by habiibullahm** [comment]
Thanks @nobodyiam — clear, I’ll skip a duplicate implementation PR.

I’d like to help with the remaining gap: document the new User Management and Permission Management OpenAPI surfaces in `docs/en|zh/portal/apollo-open-api-platform.md` (including `ManageUsers`, `/openapi/v1/users`, role grant/revoke, and the app-owner vs role boundary).

If that sounds useful, I can open a docs PR against `master`. Happy to also validate the SSO/ops flow on current `master` and note findings on #5099 / #5265.

**Comment by nobodyiam** [comment]
Yes, a focused documentation PR against `master` would be useful. The current English and Chinese Open API guides describe consumer tokens and user access tokens, but they do not yet explain `ManageUsers`, `/openapi/v1/users`, or the Permission Management role query/grant/revoke flows.

Please keep the PR as an integration guide rather than duplicating the entire OpenAPI contract. Suggested scope:

- Update both `docs/en/portal/apollo-open-api-platform.md` and `docs/zh/portal/apollo-open-api-platform.md`, and mark these APIs as available starting with Apollo 3.0.0.
- Document the authorization boundaries:
  - consumer tokens need the explicit `ManageUsers` permission for user management;
  - user access tokens need the owning user's current `ManageUsers` permission plus the `user:manage` operation;
  - role management requires the owning user's app permission plus `app:manage-role` and the applicable resource scopes.
- Add one user-search example and one role query/grant/revoke example. Consumer-token mutations require a valid `operator`; user-token requests derive the operator from the token owner.
- State that user search/get uses the configured `UserService`, while create/update/enable/disable is only supported by the built-in `SpringSecurityUserService`.
- Keep the app-owner boundary explicit: app owner is application metadata updated through `PUT /openapi/v1/apps/{appId}`; administrator, modify, and release permissions use the Permission Management APIs.
- Link to the canonical [`apollo-openapi` v0.3.10 contract](https://github.com/apolloconfig/apollo-openapi/blob/v0.3.10/apollo-openapi.yaml) for the complete paths, parameters, and schemas instead of copying every operation into both Markdown files.

For `roleType` values or missing parameter/schema descriptions, `apollo-openapi.yaml` is the source of truth. If the contract needs improvement, please make that a separate PR in `apolloconfig/apollo-openapi` rather than defining a second contract only in the guide.

Validation of the SSO/ops workflow on current `master` would also be helpful, but it does not need to block the documentation PR. One bilingual docs PR is enough.

**Comment by habiibullahm** [comment]
Thanks @nobodyiam — clear guidance. I’ll take the bilingual documentation PR against `master`.

Plan:
- Update `docs/en/portal/apollo-open-api-platform.md` and `docs/zh/portal/apollo-open-api-platform.md`
- Cover User Management (`ManageUsers`, `/openapi/v1/users`) and Permission Management (role query/grant/revoke), marked as available from Apollo 3.0.0
- Document auth boundaries for consumer tokens vs user access tokens, including `operator` handling
- Add one user-search example and one role query/grant/revoke example
- Note `UserService` (search/get) vs `SpringSecurityUserService`-only mutations, and keep the app-owner (`PUT /openapi/v1/apps/{appId}`) vs Permission Management boundary explicit
- Link to the canonical [`apollo-openapi` v0.3.10](https://github.com/apolloconfig/apollo-openapi/blob/v0.3.10/apollo-openapi.yaml) contract instead of duplicating paths/schemas

I’ll keep contract/`roleType` fixes out of this PR (separate `apollo-openapi` PR if needed). SSO/ops validation on current `master` is optional and won’t block the docs PR.

Claiming this docs work now — will open one PR when ready.
